### DISCUSSION TITLE
how do you send json with swr

### DISCUSSION THREAD

**Discussion by aliber4079** [vercel/swr#2593] [General]
all the examples in https://swr.vercel.app/docs/ show hitting a url but nothing in the docs shows how to send json.

**Comment by aliber4079** [comment]
oops I probably want useEffect instead
https://nextjs.org/docs/basic-features/data-fetching/client-side

**Comment by AbhayaShankar** [comment]
I think you could perform similar to this... We have to use fetcher function in this.
`fetcher` function that sends a `JSON` payload using the **POST** method and sets the Content-Type header to application/json. We then use the useSWR hook to fetch data from the /api/data endpoint using our fetcher function.

``` javascript
import useSWR from 'swr';

const fetcher = async (url, data) => {
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(data),
    headers: {
      'Content-Type': 'application/json'
    }
  });

  if (!response.ok) {
    throw new Error('Failed to fetch data');
  }

  return response.json();
}

const MyComponent = () => {
  const { data, error } = useSWR('/api/data', fetcher);

  if (error) return <div>Error: {error.message}</div>;
  if (!data) return <div>Loading...</div>;

  return (
    <div>
      {data.message}
    </div>
  );
}
```
